formats/src/main/java/io/parsingdata/metal/format/ZIP.java

```diff
@@ -46,38 +46,53 @@
  */
 public final class ZIP {
 
+    public static final String CRC_32 = "crc32";
+    public static final String COMPRESSED_SIZE = "compressedsize";
+    public static final String FILE_NAME_SIZE = "filenamesize";
+    public static final String EXTRA_FIELD_SIZE = "extrafieldsize";
+    public static final String EXTRACT_VERSION = "extractversion";
+    public static final String BIT_FLAG = "bitflag";
+    public static final String COMPRESSION_METHOD = "compressionmethod";
+    public static final String LAST_MOD_TIME = "lastmodtime";
+    public static final String LAST_MOD_DATE = "lastmoddate";
+    public static final String UNCOMPRESSED_SIZE = "uncompressedsize";
+    public static final String FILE_NAME = "filename";
+    public static final String COMPRESSED_DATA = "compresseddata";
+    public static final String DIR_SIGNATURE = "dirsignature";
+    public static final String EXTRA_FIELD = "extrafield";
```

Comment:
Not sure if this is really "better"... Sonar might be happy, but it's not something we typically do when we use Metal in our production code. By doing this in our examples, it looks like we promote this pattern.

On the other hand, it does show a "weakness" in our internal DSL: stringly typed field names.

Reply:
I agree. I tend to think this is actually a good pattern. Not for all values but specifically to show when a value is used multiple times.